### PR TITLE
Add dynamo smoke tests to CI

### PR DESCRIPTION
**Pull request by williamwen42**
Add dynamo smoke tests to CI, which checks for python/torch/cuda versions and runs simple dynamo examples on a few backends, including inductor. Smoke tests will run on dynamo and inductor shards.

### CODE REVIEW

File: tools/dynamo/verify_dynamo.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ def check_cuda():
     if not torch.cuda.is_available():
         return None
 
-    torch_cuda_ver = packaging.version.parse(torch.version.cuda)
+    torch_cuda_ver = packaging.version.parse(str(torch.version.cuda))
```
Comment (malfet) [inline tools/dynamo/verify_dynamo.py:61]: Why this change is necessary?

**Review by malfet — COMMENTED** [review]
LGTM, but please expand PR description to explain what is included in those smoke tests

File: tools/dynamo/verify_dynamo.py (hunk 1 above)
Comment (williamwen42) [reply tools/dynamo/verify_dynamo.py:61]: This fixed an error on one of the machines when I tried to run dynamo tests in many places. I restricted where dynamo tests are run now, so this change may no longer be necessary.

**Review by malfet — APPROVED** [review]
